libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx

```diff
@@ -0,0 +1,437 @@
+        try {
+            const attributes = getBackend().workspace(props.workspace).attributes();
+            const displayForm = await attributes.getAttributeDisplayForm(getObjRef());
+            const attribute = await attributes.getAttribute(displayForm.attribute);
+
+            setTitle(attribute.title);
+            setIsLoading(false);
+        } catch (error) {
+            setTitle("");
+            setError(error);
+            setIsLoading(false);
+
+            props.onError(error);
+        }
+    };
+
+    const loadElements = async (offset: number, limit: number) => {
+        const { workspace } = props;
+
+        setIsLoading(true);
+
+        const newElements = await getBackend()
+            .workspace(workspace)
+            .attributes()
+            .elements()
+            .forDisplayForm(getObjRef())
+            .withOptions({
+                ...(searchString ? { filter: searchString } : {}),
+            })
+            .withOffset(offset)
+            .withLimit(limit)
+            .query();
+
+        const mergedValidElements = mergeElementQueryResults(validElements, newElements);
+        const { items } = mergedValidElements;
+
+        // make sure that selected items have both title and uri, otherwise selection in InvertableList won't work
+        // TODO we could maybe use the InvertableList's getItemKey and just use title or uri for example
+        const updatedSelectedItems = updateSelectedOptionsWithData(selectedFilterOptions, items);
+        const updatedPrevSelectedItems = updateSelectedOptionsWithData(prevSelectedFilterOptions, items);
+
+        setSelectedFilterOptions(updatedSelectedItems);
+        setPrevSelectedFilterOptions(updatedPrevSelectedItems);
+        setIsLoading(false);
+        setValidElements(searchString ? newElements : mergedValidElements);
+        setTotalCount(firstLoad ? items.length : totalCount);
+        setFirstLoad(false);
+        setFilterOptions(items as Array<IAttributeElement>);
+    };
+
+    const updateSelectedOptionsWithData = (
+        selection: Array<Partial<IAttributeElement>>,
+        items: AttributeListItem[],
+    ): Array<IAttributeElement> => {
+        const nonEmptyItems = items.filter(isNonEmptyListItem);
+        const createFullItem = (item: Partial<IAttributeElement>): IAttributeElement => ({
+            uri: item.uri ?? "",
+            title: item.title ?? "",
+        });
+
+        return selection.map((selectedItem) => {
+            const foundItem = nonEmptyItems.find(
+                (item) =>
+                    (selectedItem.uri && item.uri === selectedItem.uri) ||
+                    (selectedItem.title && item.title === selectedItem.title),
+            );
+            return foundItem || createFullItem(selectedItem);
+        });
+    };
+
+    /**
+     * getters
+     */
+    const getBackend = () => {
+        return props.backend.withTelemetry("AttributeFilter", props);
+    };
+
+    const getObjRef = (): ObjRef => {
```

Comment:
This is verbatim the same as in AttributeFilter, please create an util function and unify this.